### PR TITLE
Do not apply proxy settings for JWTProxy container;

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
When workspace proxy is set, this setting applied to all containers of the workspace, including JWTProxy.  So when JWTProxy tries to access the upstream endpoint (which is always in internal form, e.g. `http://servers7ggiqm3-che-machine-execs37:4444`) it fails. This PR prevents of adding proxy settings to the JWTProxy containers, which is OK since JWTProxy never does any outbound requests.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14052

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/ProxySettingsProvisioner.java [hunk 1]
```diff
@@ -60,13 +61,15 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
...
-            .getContainers()
-            .forEach(
-                container ->
-                    proxyEnvVars.forEach((k, v) -> container.getEnv().add(new EnvVar(k, v, null))));
-      }
+      k8sEnv
+          .getPodsData()
+          .entrySet()
+          .stream()
+          .filter(entry -> !entry.getKey().equals(JWT_PROXY_POD_NAME))
```
Comment (sleshchenko) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/ProxySettingsProvisioner.java:70]: 1. Please consider adding one like comment why here we ignore JWT_PROXY pod.

2. Please consider adding test case for this.

**Review by sleshchenko — APPROVED** [review]
LGTM

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/ProxySettingsProvisioner.java (hunk 1 above)
Comment (mshaposhnik) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/ProxySettingsProvisioner.java:70]: Why i am always modifying classes which doesn;t have tests at all :rofl: ?

Comment (sleshchenko) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/ProxySettingsProvisioner.java:70]: Cause you're a lucky man! :rofl:

**Review by amisevsk — APPROVED** [review]
LGTM

**Comment by mshaposhnik** [comment]
ci-test

**Comment by mshaposhnik** [comment]
ci-build

**Review by metlos — APPROVED** [review]
LGTM

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/2217/) doesn't show any regression against this Pull Request.

**Pull request merged**